src/TrixiShallowWater.jl

```diff
@@ -1,10 +1,15 @@
 module TrixiShallowWater
-
+# Decide between using Trixi: Trixi, import Trixi or using Trixi?
 using Trixi: Trixi
```

Comment:
You would like to reuse a lot of stuff from Trixi.jl I guess? So what about

```julia
using Reexport: @reexport 
@reexport using Trixi
```

? If you do that, you don't need

```julia
using StaticArrays: SVector
```

below anymore and can remove StaticArrays.jl from your Project.toml. You could also remove all `using Trixi` statements.

Reply:
In order to test for name conflicts when using `@reexport`, we could introduce an additional submodule `Namespace` that contains the names exported by TrixiShallowWater.jl and then `@reexport` the new module:
```julia
module TrixiShallowWater
using Reexport: @reexport
@reexport using Trixi
using MuladdMacro: @muladd
using Static: True, False

include("equations/equations.jl")

# export types/functions that define the public API of TrixiShallowWater.jl
module Namespace
    export ShallowWaterEquationsWetDry1D
    # TODO: These function are currently exported by Trixi.jl. Needs to be uncommented when removed from Trixi.jl
    #export hydrostatic_reconstruction_chen_noelle, flux_nonconservative_chen_noelle, min_max_speed_chen_noelle
end
@reexport using .Namespace

end
```

However, this introduces some additional complexity and I think it can be confusing why we have this additional module. Overall, it comes down to a decision between:

- Not reexporting `using Trixi` (current version)
- Use `@reexport` and remove the nameconflict test
- Use `@reexport` and introduce the submodule

I would prefer the 1st option with the additional `using Trixi` as I think it keeps the code easy to understand and we can keep the nameconflict test. We could also easily change this at a later time as the other options so far only require changes in the `TrixiShallowWater.jl` file.

@andrewwinters5000, @ranocha what do you think about this?

Reply:
A while ago we discussed whether it might be better to not re-export everything from Trixi.jl to avoid the fact that this would somehow mean that we would have to make a breaking release in downstream packages such as TrixiSW.jl whenever there's a breaking release in Trixi.jl.

Was this argument refuted or did our perspective change in another way?

Maybe we need to properly define the relationship between the two packages (and I assume that we will use this as a blueprint for future subpackages): Is it a `TrixiSW <: Trixi` relationship or more of a "Trixi is a dependency of TrixiSW" relationship?

IMHO, requiring users to generally execute `using Trixi, TrixiSW` whenever they want to reuse functionality provided by `Trixi` (such as `trixi_include`). It also makes explicit that these are two separate packages, and that we do not guarantee that you can always do everything that's possible with Trixi also with TrixiSW.

But maybe this is a general discussion we should have in an issue and not in a PR convo :-)

Reply:
Well, we will stall have `DGSEM` etc. from Trixi.jl - so people always need to use both Trixi.jl and TrixiShallowWater.jl.
I can see arguments for both options - reexporting Trixi.jl or not. If we do not reexport stuff from Trixi.jl, it would make things a bit simpler whenever Trixi.jl has a breaking release. So let's go with option 1?

Reply:
Like @ranocha said (and @patrickersing and I discussed earlier today) there are arguments both ways for the "best" way to do this. In the end, either strategy seem to be equivalent from a user perspective. These decisions only really affect the developer end of things. So we can, ultimately, decide on a paradigm we find easiest to work with.

From the users end, the packages always need to be used together. I like option 1 where an elixir would have the first few lines be
```julia
using OrdinaryDiffEq
using Trixi
using TrixiShallowWater
```
this makes it very explicit.

Reply:
Okay, then since everyone seems to be okay with it, I would say we stick with option 1 for now and keep the current implementation.